### ISSUE TITLE
xe^x atom

### ISSUE THREAD

**Issue by SteveDiamond** [cvxpy/cvxpy#1517]
**Is your feature request related to a problem? Please describe.**
We should add an atom for the function xe^x.

**Describe the solution you'd like**
xe^x is convex when x >= 0 (actually for x >= -2). A simple approach would be to take the (non-DCP) rewrite -entr(exp(x)) and package it as a class so we can define the convexity properties. This works because -entr is increasing for arguments >= 1. I'm open to other approaches though.

**Describe alternatives you've considered**
Edit: The exponential cone proposal below is better.

**Additional context**
N/A

**Comment by parthb83** [comment]
Hey @SteveDiamond! I'm looking forward to work on this issue. Kindly guide me on the process on how to solve this issue.

**Comment by SteveDiamond** [comment]
Hi @parthb83, glad to hear you're interested! There are two steps to this issue:

1. Add a new atom in ``cvxpy/atoms/elementwise``. You can follow the structure here: https://github.com/cvxpy/cvxpy/blob/master/cvxpy/atoms/elementwise/exp.py
2. Add a rewriting in ``cvxpy/reductions/dcp2cone/atom_canonicalizers``. You can follow the structure here: https://github.com/cvxpy/cvxpy/blob/master/cvxpy/reductions/dcp2cone/atom_canonicalizers/exp_canon.py. Except you should use the rewriting:
```
x = cp.Variable(nonneg=True)
u = cp.Variable(nonneg=True)
t = cp.Variable(nonneg=True)

constraints = [cp.constraints.ExpCone(u,x,t),
                        u >= cp.power(x,2),
                        x >= 2]
return t, constraints
```

You should add extensive tests, mostly in ``cvxpy/tests/test_atoms.py`` but also in ``cvxpy/tests/test_constant_atoms.py`` and maybe in ``cvxpy/tests/test_problem.py``. Feel free to start with a simple work in progress PR and we can guide you.

**Comment by parthb83** [comment]
Hey @SteveDiamond! So I made the necessary files and changes as you laid down in the above comment. I have named the new atom function as `xexp()`. You can review the files in PR: #1641.

Can you please guide me on how to write unit test for the new atom function? I'm pretty novice in regards to writing the unit tests and any help would be much appreciated :+1:.

**Comment by rileyjmurray** [comment]
Resolved by #1641.
